modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -150,22 +143,18 @@ public void testEndpointReturnsProperSchema() throws Exception {
 		APIApplicationPublisherUtil.publishApplications(
 			_apiApplicationPublisher, _apiApplication1);
 
-		String endpointPath =
-			_apiApplication1.getBaseURL() + _API_APPLICATION_PATH_1;
-
-		HttpURLConnection httpURLConnection = _createHttpURLConnection(
-			endpointPath, Http.Method.GET);
+		JSONObject jsonObject = HTTPTestUtil.invoke(
+			null, _apiApplication1.getBaseURL() + _API_APPLICATION_PATH_1,
+			Http.Method.GET);
 
-		httpURLConnection.connect();
-
-		String actual = StringUtil.removeSubstrings(
-			StringUtil.read(httpURLConnection.getInputStream()),
-			StringPool.NEW_LINE, StringPool.SPACE);
-
-		String expected = String.format(
-			"[{\"name\":\"%s\"}]", _OBJECT_FIELD_VALUE);
-
-		Assert.assertEquals(expected, actual);
+		JSONAssert.assertEquals(
```

Comment:
Maybe better to validate the whole page instead (with items, page number, etc)

Reply:
Edit: I didn't realise you were testing in another test, but maybe it is also good to include it. Something like:

```
		JSONAssert.assertEquals(
			JSONUtil.put(
				"items", JSONUtil.put(
					JSONUtil.put("name", _OBJECT_FIELD_VALUE)
				)
			).put(
				"lastPage", 1
			).put(
				"page", 1
			).put(
				"pageSize", 20
			).put(
				"totalCount", 1
			).toString(),
			jsonObject.toString(),
			JSONCompareMode.LENIENT
```

Reply:
Done